iseq.c

```diff
@@ -1391,6 +1391,47 @@ iseqw_s_compile(int argc, VALUE *argv, VALUE self)
     return iseqw_new(rb_iseq_compile_with_option(src, file, path, line, opt));
 }
 
+static VALUE
+iseqw_s_compile_prism_compile(pm_parser_t *parser, VALUE opt, rb_iseq_t *iseq, VALUE file, VALUE path, int first_lineno) {
+    pm_node_t *node = pm_parse(parser);
+    pm_line_column_t start_line_column = pm_newline_list_line_column(&parser->newline_list, node->location.start);
+    pm_line_column_t end_line_column = pm_newline_list_line_column(&parser->newline_list, node->location.end);
+
+    rb_compile_option_t option;
+    make_compile_option(&option, opt);
+
+    prepare_iseq_build(
+        iseq,
+        rb_fstring_lit("<compiled>"),
+        file,
+        path,
+        first_lineno,
+        &(rb_code_location_t) {
+            .beg_pos = {
+                .lineno = (int) start_line_column.line,
+                .column = (int) start_line_column.column
+            },
+            .end_pos = {
+                .lineno = (int) end_line_column.line,
+                .column = (int) end_line_column.column
+            },
+        },
+        -1,
+        NULL,
+        0,
+        ISEQ_TYPE_TOP,
+        Qnil,
+        &option
+    );
+
+    pm_scope_node_t scope_node;
+    pm_scope_node_init(node, &scope_node, NULL, parser);
+    rb_iseq_compile_prism_node(iseq, &scope_node, parser);
+
+    finish_iseq_build(iseq);
+    pm_node_destroy(parser, node);
```

Comment:
Also, should this not be returning a `VALUE`? Surprised there's no compilation warning